bundles/org.openhab.core.storage.json/src/main/java/org/openhab/core/storage/json/internal/JsonStorageService.java

```diff
@@ -51,7 +53,8 @@ public class JsonStorageService implements StorageService {
     /**
      * Contains a map of needed migrations, key is the storage name
      */
-    private static final Map<String, List<TypeMigrator>> MIGRATORS = Map.of();
+    private static final Map<String, List<TypeMigrator>> MIGRATORS = Map.of( //
+            "org.openhab.core.thing.Thing", List.of(new BridgeImplTypeMigrator(), new ThingImplTypeMigrator()));
```

Comment:
It would be more modular when the migrators are part of the bundles that use the storage (like `org.openhab.core.thing` in this case). That way non-OHC bundles can also migrate their data. :slightly_smiling_face:

Reply:
Yes, but then we would have to draw in a dependency to org.openhab.core.storage.json to every bundle that uses migrations. If we keep the migrators in the storage bundle, other bundles can stay agnostic of the storage and one could use e.g. MapDB instead of JSON storage.